src/Operation/Filter.php

```diff
@@ -26,23 +26,49 @@ final class Filter extends AbstractOperation
     /**
      * @pure
      *
-     * @return Closure(callable(T , TKey, Iterator<TKey, T>): bool ...): Closure (Iterator<TKey, T>): Iterator<TKey, T>
+     * @return Closure(callable(T , TKey, Iterator<TKey, T>): bool ...): Closure (Iterator<TKey, T>): Generator<TKey, T>
      */
     public function __invoke(): Closure
     {
         return
             /**
              * @param callable(T, TKey, Iterator<TKey, T>): bool ...$callbacks
              *
-             * @return Closure(Iterator<TKey, T>): Iterator<TKey, T>
+             * @return Closure(Iterator<TKey, T>): Generator<TKey, T>
              */
             static fn (callable ...$callbacks): Closure =>
                 /**
                  * @param Iterator<TKey, T> $iterator
                  *
-                 * @return Iterator<TKey, T>
+                 * @return Generator<TKey, T>
                  */
-                static function (Iterator $iterator) use ($callbacks): Iterator {
+                static function (Iterator $iterator) use ($callbacks): Generator {
+                    // TODO: Find a way to avoid repeating this everywhere.
```

Comment:
I just noticed this pattern in the other operations now that you commented this 😄.

This can't be an operation by itself so where could we put it 🤔. Maybe in `AbstractOperation` or as a trait?

Reply:
Dunno yet, but it's going to be a follow up for sure.

Reply:
I was going to bring up the fact that we should aim to look at some of the TODOs in the codebase 😁